Formula/clozure-cl.rb

```diff
@@ -0,0 +1,67 @@
+class ClozureCl < Formula
+  desc "Common Lisp implementation with a long history"
+  homepage "https://ccl.clozure.com"
+  url "https://github.com/Clozure/ccl/archive/v1.11.5.tar.gz"
+  sha256 "07c7e05c1d50ccbf1f7602a1d436b6d6da5dcda7656b89b74daa4cf833fc7929"
+  head "https://github.com/Clozure/ccl.git"
+
+  depends_on :xcode => :build
+
+  resource "bootstrap" do
+    url "https://github.com/Clozure/ccl/releases/download/v1.11.5/ccl-1.11.5-darwinx86.tar.gz"
+    sha256 "5adbea3d8b4a2e29af30d141f781c6613844f468c0ccfa11bae908c3e9641939"
+  end
+
+  def install
+    # Unfortunately, this program is a bit finicky about which architecture, or
+    # specifically which bitness the machine is. So that's what all this
+    # business is with `MacOS.prefer_64_bit?`
+
+    heap    = MacOS.prefer_64_bit? ? "dx86cl64.image" : "dx86cl.image"
+    headers = MacOS.prefer_64_bit? ? "darwin-x86-headers64" : "darwin-x86-headers"
+
+    tmpdir = Pathname.new(Dir.mktmpdir)
+    tmpdir.install resource("bootstrap")
+    buildpath.install tmpdir/heap
+    buildpath.install tmpdir/headers
+
+    if MacOS.prefer_64_bit?
+      cd "lisp-kernel/darwinx8664" do
```

Comment:
This could be simplified a bit by adding a `build_dir` variable above, like `heap` and `headers`:

```ruby
build_dir = MacOS.prefer_64_bit? "lisp-kernel/darwinx8664" : "lisp-kernel/darwinx8632"

cd build_dir do
  system "make"
end
```

Although most ideal of all would be some kind of config-y hash containing all the products of these `prefer_64_bit?` checks.

Reply:
@woodruffw Great idea with the `build_dir` variable. I created some more variables to consolidate these checks, although I couldn't think of a way to write a "config-y hash" as you say. I didn't think I could come up with such a hash that would be cleaner than what I've already got, but feel free to improve upon it if you wish.